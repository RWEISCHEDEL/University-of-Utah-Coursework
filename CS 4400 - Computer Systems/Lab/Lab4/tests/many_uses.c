extern int a[];
int b[] = {1, 2, 3};
int c, d;

int e(int i);

int f(int i) {
  return e(a[b[i]]);
}

int g(int i) {
  c = i;
  return b[d];
}

__attribute((__noinline__)) static int j(int i)
{
  return a[i];
}

int h(int i) {
  d = i;
  return j(c);
}

__attribute((__noinline__)) int k(int i)
{
  return a[i];
}

int m(int i) {
  return k(i);
}
