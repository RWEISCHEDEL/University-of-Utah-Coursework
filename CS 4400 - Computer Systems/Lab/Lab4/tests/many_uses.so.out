f
  b
  a
  (e)
g
  c
  d
  b
m
  (k)
h
  d
  c
  a
k
  a
